pkg/extensions/tbtc/recovery/recovery.go

```diff
@@ -0,0 +1,174 @@
+			err,
+		)
+	}
+	pkhAddress := pubKeyAddress.AddressPubKeyHash()
+
+	script, err := txscript.PayToAddrScript(pkhAddress)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"error deriving p2wpkh scriptCode from public key: [%s]",
+			err,
+		)
+	}
+	if len(script) != 25 {
+		return nil, fmt.Errorf(
+			"error deriving p2wpkh scriptCode from public key: [unexpected scriptCode length: %v]",
+			len(script),
+		)
+	}
+
+	// End goal here is a scriptCode that looks like
+	// 0x1976a914{20-byte-pubkey-hash}88ac . 0x19 should be the length of the
+	// script.
+	return append([]byte{byte(len(script))}, script...), nil
+}
+
+// ConstructUnsignedTransaction produces an unsigned transaction
+func ConstructUnsignedTransaction(
+	previousOutputTransactionHashHex string,
+	previousOutputIndex uint32,
+	previousOutputValue int64,
+	feePerVbyte int64,
+	recipientAddresses []string,
+	chainParams *chaincfg.Params,
+) (*wire.MsgTx, error) {
+	// If the previous output transaction hash is passed as a []byte, can use
+	// chainhash.NewHash.
+	previousOutputTransactionHash, err := chainhash.NewHashFromStr(
+		previousOutputTransactionHashHex,
+	)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"error extracting outpoint transaction hash: [%s]",
+			previousOutputTransactionHash,
+		)
+	}
+
+	// The witness signature field is the DER signature followed by the hash type.
+	// We write a dummy signature with 73 0 bytes. DER signatures vary in encoding
+	// between 71, 72, and 73 bytes, so we choose the longest for fee purposes.
+	// We then add one more dummy byte for the SigHashType for a total of 74 bytes.
+	dummySignatureForWitness := bytes.Repeat([]byte{0}, 74)
+
+	// The compressed public key requires 33 bytes.
+	dummyCompressedPublicKeyForWitness := bytes.Repeat([]byte{0}, 33)
+
+	tx := wire.NewMsgTx(wire.TxVersion)
+	txIn := wire.NewTxIn(
+		wire.NewOutPoint(previousOutputTransactionHash, previousOutputIndex),
+		nil, // scriptSig is empty here
+		[][]byte{
+			dummySignatureForWitness,
+			dummyCompressedPublicKeyForWitness,
+		},
+	)
+	txIn.Sequence = 0
+	tx.AddTxIn(txIn)
+
+	for _, recipientAddress := range recipientAddresses {
+		address, err := btcutil.DecodeAddress(recipientAddress, chainParams)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"error decoding output address [%s]: [%s]",
+				recipientAddress,
+				err,
+			)
+		}
+		outputScript, err := txscript.PayToAddrScript(address)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"error constructing script from output address [%s]: [%s]",
+				recipientAddress,
+				err,
+			)
+		}
+
+		tx.AddTxOut(wire.NewTxOut(
+			int64(0), // value is filled in after fee is computed below
+			outputScript,
+		))
+	}
+
+	// Compute weight and vsize per [BIP141], except vsize is truncated
+	// instead of rounded up, then compute the final fee and set the
+	// per-recipient value. Could result in a fractionally low fee.
+	weight := tx.SerializeSizeStripped()*3 + tx.SerializeSize()
+	vsize := weight / 4
```

Comment:
I'm wondering if it would be worth using a [`GetTxVirtualSize`function from btcd](https://github.com/btcsuite/btcd/blob/f86ae60936d7134fad9bf0ab1b3a1c261abe79b7/mempool/policy.go#L375) to calculate the virtual size? The only difference I see is that it rounds up the size, which is consistent with what was mentioned in BIP141:
> Virtual transaction size is defined as Transaction weight / 4 (rounded up to the next integer).

Is there any reason we don't want to round it up?

CC: @Shadowfiend 

```suggestion
	vsize := mempool.GetTxVirtualSize(btcutil.NewTx(tx))
```

Reply:
I'm going to go ahead and push the commit that uses this, since I think it's the standard route, and then we can back it out if that isn't the case